src/HealthGPS/person.cpp

```diff
@@ -30,34 +41,24 @@ unsigned int Person::time_of_migration() const noexcept { return time_of_migrati
 
 bool Person::is_active() const noexcept { return is_alive_ && !has_emigrated_; }
 
-double Person::get_risk_factor_value(const core::Identifier &key) const noexcept {
-    // Static properties
+double Person::get_risk_factor_value(const core::Identifier &key) const {
     if (current_dispatcher.contains(key)) {
+        // Static properties
         return current_dispatcher.at(key)(*this);
-    }
-
-    // Dynamic properties
-    if (risk_factors.contains(key)) {
+    } else if (risk_factors.contains(key)) {
+        // Dynamic properties
         return risk_factors.at(key);
+    } else {
+        throw std::out_of_range("Risk factor not found: " + key.to_string());
     }
```

Comment:
This is more of a personal thing, but I'd generally write something like this as:

```cpp
if (first_thing) {
    return 1;
}
if (second_thing) {
    return 2;
}

throw std::exception("bad thing happened!");
```

I find the cognitive load is a bit less than when you have a bunch of `else`s.

Reply:
Huh. Interesting little differences. I actually find the explicitness helpful. Not married to either way though, if it helps the reviewers.

Reply:
Totally up to you. This one is mostly a matter of taste, I think.